qemu/qmp.go

```diff
@@ -1158,6 +1158,30 @@ func (q *QMP) ExecutePCIVFIOMediatedDeviceAdd(ctx context.Context, devID, sysfsd
 	return q.executeCommand(ctx, "device_add", args, nil)
 }
 
+// isSocketIDSupported returns if the cpu driver supports the socket id option
+func isSocketIDSupported(driver string) bool {
+	if driver == "host-s390x-cpu" || driver == "host-powerpc64-cpu" {
+		return false
+	}
+	return true
+}
+
+// isThreadIDSupported returns if the cpu driver supports the thread id option
+func isThreadIDSupported(driver string) bool {
+	if driver == "host-s390x-cpu" || driver == "host-powerpc64-cpu" {
+		return false
+	}
+	return true
+}
+
+// isDieIDSupported returns if the cpu driver and the qemu version support the die id option
+func (q *QMP) isDieIDSupported(driver string) bool {
+	if q.version.Major > 4 || (q.version.Major == 4 && q.version.Minor >= 1) && driver == "qemu64-x86_64-cpu" {
```

Comment:
in my system the driver is `host-x86_64-cpu`  :thinking:

Reply:
yeah, I had that previously but then I saw the test and [here](https://wiki.qemu.org/Features/CPUHotplug)

Reply:
heh - I think `qemu64-x86_64-cpu` was used just for testing :smile:

Reply:
@sboeuf Any idea what the correct cpu driver to use here is?  Perhaps we need to check for both.  Basically, we only want die-id to be specified on x86 systems.

Reply:
ok changed back to host-x86_64-cpu

Reply:
@markdryan  `host-x86_64-cpu` is correct

```
$ qemu-system-x86_64 -device host-x86_64-cpu,help
host-x86_64-cpu options:
  min-xlevel2=<uint32>
  amd-ssbd=<bool>
  vendor=<string>
...
```

Reply:
and the qmp command `query-hotpluggable-cpus` returns this:

```
{"props": {"core-id": 0, "thread-id": 0, "node-id": 0, "socket-id": 7}, "vcpus-count": 1, "type": "host-x86_64-cpu"}
```